src/utils/LoggingUtil.ts

```diff
@@ -0,0 +1,32 @@
+import { TEST_STATUS, TEST_RESULT } from "../assets/Enums";
+import { Defect, TestType } from "../models/ITestResult";
+import { ITestResultPayload } from "../models/ITestResultPayload";
+
+export class LoggingUtil {
+    private static readonly reportingDeficiencyRef = ["8.1.i", "8.1.j.i", "8.1.j.ii", "8.1", "8.2"];
+    /**
+     * Function to log certain defects and advisory silently
+     */
+    public static logDefectsReporting(testResult: ITestResultPayload): void {
+
+        if (testResult.testStatus === TEST_STATUS.CANCELLED) { return; }
+
+        testResult.testTypes.forEach((testType: TestType) => {
+            if (testType.testResult !== TEST_RESULT.ABANDONED) {
+                testType.defects.forEach((defect: Defect) => {
+                    if (LoggingUtil.reportingDeficiencyRef.includes(defect.deficiencyRef)) {
+                        const logObject = {vin: testResult.vin,
+                            vrm: testResult.vrm,
+                            additionalNotesRecorded: testType.additionalNotesRecorded,
+                            deficiencyRef: defect.deficiencyRef,
+                            deficiencyCategory: defect.deficiencyCategory,
+                            defectNotes: defect?.additionalInformation.notes,
+                            ...defect?.additionalInformation?.location
```

Comment:
can we have undefined values for location? if it is the case do we need a default placeholder value as if I understand the ticket correctly this props is only present for 8.1 & 8.2 ?
Or maybe it is something we can double check with the business

Reply:
yes we can have undefined values but that is not a problem since this information will be extracted from CloudWatch in a structured CSV file